tkg/client/cluster.go

```diff
@@ -952,6 +956,44 @@ func (c *TkgClient) ValidateKubeVipLBConfiguration(clusterRole string) error {
 	return nil
 }
 
+// ValidateAviMaxAllowedClusterNameLength validates if AVI enabled cluster's name has no more than 25 characters
+func (c *TkgClient) ValidateAviMaxAllowedClusterNameLength(clusterRole string, options *CreateClusterOptions, regionalClusterClient clusterclient.Client) error {
+	aviLabels, _ := c.TKGConfigReaderWriter().Get(constants.ConfigVariableAviLabels)
+	kvlbEnabled, _ := c.TKGConfigReaderWriter().Get(constants.ConfigVariableKubevipLoadbalancerEnable)
+
+	// skip when:
+	// 1. cluster is management cluster, that check has already been done in ConfigureAndValidateAviConfiguration func
+	// 2. workload cluster doesn't enable avi
+	// 3. workload cluster is not clusterclass based cluster
+	if clusterRole != TkgLabelClusterRoleWorkload || kvlbEnabled == "true" {
+		return nil
+	}
+
+	mgmtClusterName, _, err := c.getRegionalClusterNameAndNamespace(regionalClusterClient)
+	if err != nil {
+		return errors.Wrap(err, "unable to get name and namespace of current management cluster")
+	}
+	akooAddonSecretValues, found, err := GetAKOOAddonSecretValues(regionalClusterClient, mgmtClusterName, true)
+	if err != nil {
+		return errors.Wrap(err, "unable to get akoo addon secret values")
+	}
+
+	if found {
+		configValues := make(map[string]interface{})
+		if err = RetrieveAKOOVariablesFromAddonSecretValues(mgmtClusterName, configValues, akooAddonSecretValues); err != nil {
+			return errors.Wrap(err, "unable to handle the akoo specific variables")
+		}
+		// mgmt avi labels is specified & workload cluster avi labels not specify, workload cluster doesn't enable Avi, skip
+		if !utils.IsAviInputEmpty(configValues[constants.ConfigVariableAviLabels]) && aviLabels == "" {
+			return nil
+		}
+		if len(options.ClusterName) > constants.AkoMaxAllowedClusterNameLen {
+			return errors.Errorf("%s contains more than %d character, which will cause AKO package deployment failure", options.ClusterName, constants.AkoMaxAllowedClusterNameLen)
+		}
+	}
+	return nil
```

Comment:
> Do we handle the case when workload cluster is not clusterclass

If cluster is not clusterclass based, akoo secret will not be found, then there will be no validation on the legacy cluster, which should be good.

> if akooAddonSecretValues not found, we should return nil or error?

We should return nil because when the mgmt cluster doesn't have AVI enabled, it should be ok to not have akoo pod in the management cluster.